src/js/components/Button/Button.js

```diff
@@ -87,13 +87,21 @@ class Button extends Component {
     const { hover } = this.state;
 
     let buttonIcon = icon;
-    // only change color if user did not specify the color themselves...
-    if (primary && icon && !icon.props.color) {
+    if (icon && !icon.props.color) {
+      let buttonIconColor;
+      if (!primary && theme.button.color){
+      buttonIconColor = normalizeColor(theme.button.color, theme);
+      }
+      else if (primary && theme.button.color) {
+          buttonIconColor = normalizeColor(theme.button.color, theme);
+        } else if (primary && !theme.button.color) {
+          buttonIconColor =
+            theme.global.colors.text[
+              isDarkBackground(this.props) ? 'dark' : 'light'
+            ];
+          }
```

Comment:
I feel like there are too many conditions and nested behavior in this block (since line #90), even if it works properly we would like to have a code that is more readable and easy to maintain. Please take some time to try to clean it up and use more simple logic.

Reply:
@ShimiSun I will work on cleaning this up!